csrc/host_ir/jit.cpp

```diff
@@ -243,49 +268,240 @@ void packOutputs(
+
+    llvm::Value* sizes_array =
+        builder_.CreateAlloca(sizes_array_type, nullptr, "sizes_array");
+    llvm::Value* strides_array =
+        builder_.CreateAlloca(strides_array_type, nullptr, "strides_array");
+
+    // Populate sizes array
+    for (size_t i = 0; i < tensor_sizes.size(); ++i) {
+      llvm::Value* gep = builder_.CreateInBoundsGEP(
+          sizes_array_type,
+          sizes_array,
+          {builder_.getInt32(0), builder_.getInt32(i)});
+      builder_.CreateStore(tensor_sizes[i], gep);
+    }
+
+    // Populate strides array
+    for (size_t i = 0; i < tensor_strides.size(); ++i) {
+      llvm::Value* gep = builder_.CreateInBoundsGEP(
+          strides_array_type,
+          strides_array,
+          {builder_.getInt32(0), builder_.getInt32(i)});
+      builder_.CreateStore(tensor_strides[i], gep);
+    }
+
+    // Convert arrays to pointers
+    llvm::Value* sizes_arg =
+        builder_.CreateBitCast(sizes_array, int64_ptr_type);
+    llvm::Value* strides_arg =
+        builder_.CreateBitCast(strides_array, int64_ptr_type);
+
+    // Create array size arguments
+    llvm::Value* shape_ndim_arg = builder_.getInt64(tensor_sizes.size());
+    llvm::Value* strides_ndim_arg = builder_.getInt64(tensor_strides.size());
+
+    // Create output tensor
+    llvm::Value* raw_tensor_ptr = builder_.CreateCall(
+        mod->getFunction(kNewTensorFuncName), {}, "out_tensor");
+
+    // Create constants for type and device from params
+    at::ScalarType data_type = data_type_to_aten(
+        allocate->buffer()->dtype() == DataType::Index
+            ? PrimDataType::Int
+            : allocate->buffer()->dtype());
+    llvm::Value* dtype_constant =
+        builder_.getInt32(static_cast<int32_t>(data_type));
+    llvm::Value* device_index_constant =
+        builder_.getInt64(Communicator::getInstance().deviceId());
+
+    // Configure output tensor
+    llvm::Function* at_empty_strided_cuda_func =
+        mod->getFunction(kAtEmptyStridedCudaWrapper);
+
+    // Call at::native::empty_strided_cuda with the computed arguments
+    builder_.CreateCall(
+        at_empty_strided_cuda_func,
+        {sizes_arg,
+         shape_ndim_arg,
+         strides_arg,
+         strides_ndim_arg,
+         dtype_constant,
+         device_index_constant,
+         raw_tensor_ptr});
+    val_to_value_[allocate->buffer()->as<Val>()] = raw_tensor_ptr;
+
+    if (isDebugDumpEnabled(DebugDumpOption::HostIrJit)) {
+      llvm::outs() << "=== LLVM IR After Generating Allocate Function ===\n";
+      mod->print(llvm::outs(), nullptr);
+    }
+  }
+
+  // Deallocation Function LLVM IR Generation
+  void handle(hir::Deallocate* deallocate) final {
+    auto mod = builder_.GetInsertBlock()->getParent()->getParent();
+    llvm::Function* delete_tensor_func =
+        mod->getFunction(kDeleteTensorFuncName);
+    builder_.CreateCall(
+        delete_tensor_func, {val_to_value_[deallocate->buffer()->as<Val>()]});
```

Comment:
```suggestion
        delete_tensor_func, {val_to_value_.at(deallocate->buffer())});
```
`.at` asserts the key exists.